tests/e2e/specs/proposal.spec.js

```diff
@@ -0,0 +1,206 @@
+    it('should connect to wallet', () => {
+      if (!networkPhrases.isLocal) {
+        cy.origin('https://wallet.agoric.app/', () => {
+          cy.visit('/');
+        });
+        cy.acceptAccess();
+        cy.origin('https://wallet.agoric.app/', () => {
+          cy.visit('/wallet/');
+
+          cy.get('input.PrivateSwitchBase-input').click();
+          cy.contains('Proceed').click();
+
+          cy.get('button[aria-label="Settings"]').click();
+
+          cy.get('#demo-simple-select').click();
+          cy.get('li[data-value="testnet"]').click();
+          cy.contains('button', 'Connect').click();
+        });
+
+        cy.acceptAccess();
+      }
+      cy.visit(`/?agoricNet=${networkPhrases.network}`);
+      cy.acceptAccess();
+    });
+
+    it('should allow gov1 to create a proposal', () => {
+      // open PSM and select token
+      cy.get('button').contains('PSM').click();
+      cy.get('button').contains('AUSD').click();
+      cy.get('button').contains(networkPhrases.token).click();
+
+      // Change mint limit and proposal time to 1 min
+      cy.get('label')
+        .contains('Set Mint Limit')
+        .parent()
+        .within(() => {
+          cy.get('input').spread(element => {
+            cy.get('input').clear().type(element.value);
+          });
+        });
+      cy.get('label')
+        .contains('Minutes until close of vote')
+        .parent()
+        .within(() => {
+          cy.get('input').clear().type(networkPhrases.minutes);
+        });
+      cy.get('[value="Propose Parameter Change"]').click();
+
+      // Submit proposal and wait for confirmation
+      cy.confirmTransaction();
+      cy.get('p')
+        .contains('sent', { timeout: 120_000 })
+        .should('be.visible')
+        .then(() => {
+          startTime = Date.now();
+        });
+    });
+
+    it('should allow gov2 to vote on the proposal', () => {
+      cy.visit(`/?agoricNet=${networkPhrases.network}`);
+
+      // Open vote, click on yes and submit
+      cy.get('button').contains('Vote').click();
+      cy.get('p').contains('YES').click();
+      cy.get('input:enabled[value="Submit Vote"]').click();
+
+      // Wait for vote to confirm
+      cy.confirmTransaction();
+      cy.get('p').contains('sent', { timeout: 120_000 }).should('be.visible');
+    });
+
+    it('should allow gov1 to vote on the proposal', () => {
+      cy.switchWallet('gov1');
+      cy.visit(`/?agoricNet=${networkPhrases.network}`);
+
+      // Open vote, click on yes and submit
+      cy.get('button').contains('Vote').click();
+      cy.get('p').contains('YES').click();
+      cy.get('input:enabled[value="Submit Vote"]').click();
+
+      // Wait for vote to confirm
+      cy.confirmTransaction();
+      cy.get('p').contains('sent', { timeout: 120_000 }).should('be.visible');
+    });
+
+    it('should wait for proposal to pass', () => {
+      // Wait for 1 minute to pass
+      cy.wait(60_000 * networkPhrases.minutes - Date.now() + startTime);
```

Comment:
this is to wait until one minute after startTime? please make a helper method to make that logic more clear and not repeated